File: cryptography-cipher/src/main/java/dev/retrotv/crypto/cipher/result/AEADResult.java
package dev.retrotv.crypto.cipher.result;

import lombok.Getter;

/**
 * AEAD 암호화 결과를 담는 클래스 입니다.
 */
@Getter
public class AEADResult extends Result {
    private final byte[] tag;

    /**
     * AEADResult 객체 생성자
     *
     * @param data 암호화 또는 복호화된 데이터
     * @param tag 인증 태그
     */
    public AEADResult(byte[] data, byte[] tag) {
        super(data);
        this.tag = tag;
    }
}

File: cryptography-cipher/src/main/kotlin/dev/retrotv/crypto/cipher/result/AEADResult.kt
package dev.retrotv.crypto.cipher.result

/**
 * AEAD 암호화 결과를 담는 클래스 입니다.
 * @property data 암호화된 데이터
 * @property tag 인증 태그
 */
class AEADResult(
    override val data: ByteArray,
    val tag: ByteArray
) : Result(data)